configure

```diff
@@ -0,0 +1,344 @@
+        -*) warning Unknown argument: "$arg" ;;
+        *=*) eval "$arg" ;;
+        *) continue ;;
+    esac
+done
+
+## Verify PLATFORM value
+if [ "$PLATFORM" != "Linux" ] && [ "$PLATFORM" != "FreeBSD" ] && \
+[ "$PLATFORM" != "OpenBSD" ] && [ "$PLATFORM" != "Darwin" ]; then
+    warning "$PLATFORM" platform is unknown!
+    sub_warning Valid Platforms are: Linux, FreeBSD, OpenBSD, Darwin
+fi
+
+## Create testfile.cc to test c++ compiler
+create_testfile_cc
+
+## Find and test C++ compiler
+info Checking C++ Compiler...
+if test -n "$CXX" ; then
+    sub_info "$CXX" is user/system prefered C++ compiler.
+    compiler="$CXX"
+elif test -n "$TARGET"; then
+    for guess in "$TARGET"-g++ "$TARGET"-clang++ "$TARGET"-c++; do
+        findcxx "$guess"
+        if test -n "$compiler"; then
+            sub_info "$compiler"
+            break # Found
+        fi
+    done
+    info Checking Host C++ Compiler...
+    if test -n "$BUILD"; then
+        for guess in "$BUILD"-g++ "$BUILD"-clang++ "$BUILD"-c++; do
+            findcxx "$guess" host
+            if test -n "$hostcompiler"; then
+                sub_info "$hostcompiler"
+                break # Found
+            fi
+        done
+    else
+        for guess in g++ clang++ c++; do
+        findcxx "$guess" host
+        if test -n "$hostcompiler"; then
+            sub_info "$hostcompiler"
+            break # Found
+        fi
+        done
+    fi
+    if test -z "$compiler" || test -z "$hostcompiler"; then
+       sub_error No C++ compiler found! # Not found
+    fi
+else
+    for guess in g++ clang++ c++; do
+        findcxx "$guess"
+        if test -n "$compiler"; then
+            sub_info "$compiler"
+            break # Found
+        fi
+    done
+    if test -z "$compiler"; then
+       sub_error No C++ compiler found! # Not found
+    fi
+fi
+cxx_works
+
+## Find depends
+if [ "$PLATFORM" = "Linux" ]; then
+    info Checking make is executable...
+    find_program make
+else
+    info Checking gmake is executable...
+    find_program gmake
+fi
+
+## Test compiler/linker supported arguments
+if test -z $NO_CXXOPTS; then
+    for argument in -std=c++11 \
+                    -Wall \
+                    -Os \
+                    -fno-plt
+    do
+        try_cxx_argument CXXOPTS $argument
+    done
+    if [ "$PLATFORM" != "FreeBSD" ]; then
+        try_cxx_argument CXXOPTS -flto
+    fi
+    if [ "$PLATFORM" != "Darwin" ]; then
+        try_cxx_argument CXXOPTS -fno-rtti
+    fi
+fi
+if [ "$PLATFORM" = "FreeBSD" ] && test -z $NO_LDFLAGS; then
+    try_ld_argument LDFLAGS -lrt
+fi
+if [ "$NO_SANITIZE" = "auto" ]; then
+    try_ld_argument SANITIZEOPTS -fsanitize=address,undefined
+elif [ "$NO_SANITIZE" = "0" ]; then
+    SANITIZEOPTS="-fsanitize=address,undefined"
+fi
+
+## Create mconfig
+rm -f testfile*
+info Creating mconfig...
+cat << EOF > mconfig
+## Auto-generated by "$0" for "$PLATFORM"
+# Installation path options.
+
+SBINDIR=$SBINDIR
```

Comment:
Note that this will fail to correctly quote the output. Eg try `--sbindir="/usr/some dir"` or some other path containing a space. There's no easy way to solve that though.

Reply:
I'm think it's easiest solution: "Please use `\` (E.g `/usr/some\ dir`)  for paths containing a space"

Reply:
yeah but if you do `./configure --sbindir="/usr/some\ dir" it won't work. The user would need to double the backslash: `./configure --sbindir="/usr/some\\ dir". Don't worry about it for now.

Reply:
Are you sure about this, I do `./configure --sbindir="/usr/some\ dir"` and it works.
> Don't worry about it for now.
Ok

Reply:
> Are you sure about this,
Actually, no, you are right. I thought that `\x` for any x would be collapsed to just `x` but this is not the case apparently.

Reply:
I mention this thing in `./configure --help`